pennylane/noise/conditionals.py

```diff
@@ -0,0 +1,458 @@
+        return False
+
+    coeffs, op_terms = lc_cop(op1).terms()
+    sprods = [_get_ops(getattr(op_term, "operands", op_term)) for op_term in op_terms]
+
+    def _lc_op(x):
+        coeffs2, op_terms2 = lc_cop(x).terms()
+        sprods2 = [_get_ops(getattr(op_term, "operands", op_term)) for op_term in op_terms2]
+        for coeff, sprod in zip(coeffs2, sprods2):
+            present, p_index = False, -1
+            while sprod in sprods[p_index + 1 :]:
+                p_index = sprods[p_index + 1 :].index(sprod) + (p_index + 1)
+                if qml.math.allclose(coeff, coeffs[p_index]):
+                    coeffs.pop(p_index)
+                    sprods.pop(p_index)
+                    present = True
+                    break
+
+            if not present:
+                break
+
+        return present
+
+    return _lc_op(op2)
+
+
+def op_in(ops):
+    """Builds a conditional as a boolean function for evaluating
+    if a given operation exist in a specified set of operations.
+
+    Args:
+        ops (str, class, Operation, Union(list[str, class, Operation])): sequence of string
+            representations, instances or classes of the operation(s).
+
+    Returns:
+        :class:`OpIn <pennylane.noise.OpIn>`: a boolean function that evaluates to ``True``, if a
+        given operation exist in a specified set of operation(s) based on a comparison of the type,
+        irrespective of their wires.
+
+    **Example**
+
+    One may use ``op_in`` with a string representation of the name of the operation:
+
+    >>> cond_func = qml.noise.op_in(["RX", "RY"])
+    >>> cond_func(qml.RX(1.23, wires=[0]))
+    True
+    >>> cond_func(qml.RZ(1.23, wires=[3]))
+    False
+    >>> cond_func([qml.RX(1.23, wires=[1]), qml.RY(4.56, wires=[2])])
+    True
+
+    Additionally, an instance of :class:`Operation <pennylane.operation.Operation>`
+    can also be provided:
+
+    >>> cond_func = qml.noise.op_in([qml.RX(1.0, "dino"), qml.RY(2.0, "rhino")])
+    >>> cond_func(qml.RX(1.23, wires=["eve"]))
+    True
+    >>> cond_func(qml.RY(1.23, wires=["dino"]))
+    True
+    >>> cond_func([qml.RX(1.23, wires=[1]), qml.RZ(4.56, wires=[2])])
+    False
+    """
+    ops = [ops] if not isinstance(ops, (list, tuple, set)) else ops
+    return OpIn(ops)
+
+
+def op_eq(ops):
+    """Builds a conditional as a boolean function for evaluating
+    if a given operation is equal to the specified operation.
+
+    Args:
+        ops (str, class, Operation): string representation, an instance or class of the operation.
+
+    Returns:
+        :class:`OpEq <pennylane.noise.OpEq>`: a boolean function that evaluates to ``True``, if a
+        given operation is equal to the specified set of operation(s) based on a comparison of
+        the type, irrespective of their wires.
+
+    **Example**
+
+    One may use ``op_eq`` with a string representation of the name of the operation:
+
+    >>> cond_func = qml.noise.op_eq("RX")
```

Comment:
not blocking
"X", "Y", "Z" works but "I" does not

Reply:
Nice catch, fixed!